File: Examples/VisualBasic/ModifyingAndConvertingImages/ConcatenatingTIFFImagesfromStream.vb
Imports System.IO
Imports Aspose.Imaging.FileFormats.Tiff

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Public Class ConcatenatingTIFFImagesfromStream
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Create instances of FileStream and initialize with Tiff images
            Dim fileStream As New FileStream(dataDir & Convert.ToString("TestDemo.tif"), FileMode.Open)
            Dim fileStream1 As New FileStream(dataDir & Convert.ToString("sample.tif"), FileMode.Open)

            ' Create an instance of TiffImage and load the destination image from filestream
            Using image As TiffImage = DirectCast(Imaging.Image.Load(fileStream), TiffImage)
                ' Create an instance of TiffImage and load the source image from filestream
                Using image1 As TiffImage = DirectCast(Imaging.Image.Load(fileStream1), TiffImage)
                    ' Create an instance of TIffFrame and copy active frame of source image and  Add copied frame to destination image
                    Dim frame As TiffFrame = TiffFrame.CopyFrame(image1.ActiveFrame)
                    image.AddFrame(frame)
                End Using
                ' Save the image with changes and Close the FileStreams
                image.Save(dataDir & Convert.ToString("ConcatenatingTIFFImagesfromStream_out.tif"))
                fileStream.Close()
                fileStream1.Close()
            End Using
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/ConcatenatingTIFFImagesfromStream.cs
using System;
using System.IO;
using Aspose.Imaging;
using Aspose.Imaging.FileFormats.Tiff;

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    public class ConcatenatingTIFFImagesfromStream
    {
        public static void Run()
        {
            Console.WriteLine("Running example ConcatenatingTIFFImagesfromStream");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Create FileStream instances and initialize them with TIFF images.
            FileStream fileStream = new FileStream(dataDir + "TestDemo.tif", FileMode.Open);
            FileStream fileStream1 = new FileStream(dataDir + "sample1.tif", FileMode.Open);

            // Load the destination image from the first filestream.
            using (TiffImage image = (TiffImage)Image.Load(fileStream))
            {
                // Load the source image from the second filestream.
                using (TiffImage image1 = (TiffImage)Image.Load(fileStream1))
                {
                    // Create a TiffFrame and copy the active frame of the source image.
                    TiffFrame frame = TiffFrame.CopyFrame(image1.ActiveFrame);

                    // Add the copied frame to the destination image.
                    image.AddFrame(frame);
                }

                // Save the image with the added frame.
                image.Save(dataDir + "ConcatenatingTIFFImagesfromStream_out.tif");

                // Close the FileStreams.
                fileStream.Close();
                fileStream1.Close();
            }

            Console.WriteLine("Finished example ConcatenatingTIFFImagesfromStream");
        }
    }
}